### PR TITLE
Adding devices support for XC Model

### PR DESCRIPTION
**Pull request by shaipranesh2**
## Description
Currently machines with 2 or more computational devices (i.e cuda, CPU) have trouble running the XC model.
Have fixed it up in this PR
Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by shaipranesh2** [comment]
@rbharath @advikavs I have made the changes for the XC model to add in devices support.
I could not run XC model for classification on deepchem/models/tests/assets/test_dftxcdata.yaml using CUDA, because I faced problems with dqc.Mol used in featurizer and loading it up with CUDA (This can be found in feat/dft_data.py).

Hence, I made "CPU" as the default device, this change allows computers with both CPU and GPU to run the tutorial unlike the previous time.

Also made minor changes to the notebook.

**Comment by shaipranesh2** [comment]
@advikavs the pbe_test failed due to a matrix shape error. I only changed the input features to 2 and now it passes the test originally written. Is this fine?

**Review by rbharath — COMMENTED** [review]
@advikavs @arunppsg Would you be able to review this PR and make comments?

File: deepchem/models/dft/dftxc.py [hunk 1]
```diff
@@ -130,7 +144,7 @@ def __init__(self,
                  n_tasks: int = 0,
                  log_frequency: int = 0,
                  mode: str = 'classification',
-                 device: Optional[torch.device] = None,
+                 device: str = "cpu",
```
Comment (arunppsg) [inline deepchem/models/dft/dftxc.py:147]: nit: any advantage of having device as `str` rather than `torch.device`?

**Review by arunppsg — COMMENTED** [review]
lgtm

File: deepchem/models/dft/dftxc.py (hunk 1 above)
Comment (shaipranesh2) [reply deepchem/models/dft/dftxc.py:147]: There seems to be no advantage, and there is some mypy issues on this. I will resolve it soon and tag for merging

**Comment by shaipranesh2** [comment]
Is there a way to test model on cuda in the CI?

**Review by rbharath — COMMENTED** [review]
@shaipranesh2 This PR is too big to merge or review as is. Can you make a smaller PR that makes device fixes in 1 file to start?

File: deepchem/data/data_loader.py [hunk 2]
```diff
@@ -1660,15 +1665,24 @@ class DFTYamlLoader(DataLoader):
 
     """
 
-    def __init__(self):
+    def __init__(self, device: Optional[torch.device] = None):
```
Comment (rbharath) [inline deepchem/data/data_loader.py:1668]: A Yaml loader should not know about a torch device

File: deepchem/data/data_loader.py [hunk 3]
```diff
@@ -1745,7 +1770,11 @@ def _featurize_shard(self, shard):
             )
         if 'weight' in shard.keys():
             weight = shard['weight']
-            x = DFTEntry.create(e_type, true_val, systems, weight)
+            x = DFTEntry.create(e_type,
```
Comment (rbharath) [inline deepchem/data/data_loader.py:1773]: A DFTEntry object should not know about the device. That is a feature of the model and not of the data

File: deepchem/feat/dft_data.py [hunk 4]
```diff
@@ -36,7 +39,8 @@ class DFTSystem():
     https://github.com/diffqc/dqc/blob/0fe821fc92cb3457fb14f6dff0c223641c514ddb/dqc/system/base_system.py
     """
 
-    def __init__(self, system: Dict):
+    def __init__(self, system: Dict, device: torch.tensor):
```
Comment (rbharath) [inline deepchem/feat/dft_data.py:42]: The DFTSystem should not know about the device
